tests/unit/s2n_client_hello_test.c

```diff
@@ -255,6 +276,28 @@ int main(int argc, char **argv)
         free(extensions_out);
         extensions_out = NULL;
 
+        /* Verify server name extension and it's length are returned correctly */
+        EXPECT_EQUAL(s2n_client_hello_get_extension_length(client_hello, S2N_EXTENSION_SERVER_NAME), server_name_extension_len);
+        EXPECT_NOT_NULL(ext_data = malloc(server_name_extension_len));
+        EXPECT_EQUAL(s2n_client_hello_get_extension_by_id(client_hello, S2N_EXTENSION_SERVER_NAME, ext_data, server_name_extension_len), server_name_extension_len);
+        EXPECT_SUCCESS(memcmp(ext_data, server_name_extension, server_name_extension_len));
+        free(ext_data);
+        ext_data = NULL;
+
+        /* Verify server name extension is truncated if extension_size > max_len */
+        EXPECT_NOT_NULL(ext_data = malloc(server_name_extension_len - 1));
+        EXPECT_EQUAL(s2n_client_hello_get_extension_by_id(client_hello, S2N_EXTENSION_SERVER_NAME, ext_data, server_name_extension_len - 1), server_name_extension_len - 1);
+        EXPECT_SUCCESS(memcmp(ext_data, server_name_extension, server_name_extension_len - 1));
+        free(ext_data);
+        ext_data = NULL;
+
+        /* Verify get extension and it's length calls for a non-existing extension type */
+        EXPECT_EQUAL(s2n_client_hello_get_extension_length(client_hello, S2N_EXTENSION_CERTIFICATE_TRANSPARENCY), 0);
+        EXPECT_NOT_NULL(ext_data = malloc(server_name_extension_len));
+        EXPECT_EQUAL(s2n_client_hello_get_extension_by_id(client_hello, S2N_EXTENSION_CERTIFICATE_TRANSPARENCY, ext_data, server_name_extension_len), 0);
+        free(ext_data);
+        ext_data = NULL;
+
```

Comment:
A bit further down in this test the connection is wiped (using ```s2n_connection_wipe```). We should check the state of the ```parsed_extensions``` array after wipe (can be added after the checks for the ```cipher_suites``` and ```extensions``` blobs).

Reply:
Fair point, added.